app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -206,6 +210,19 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
             }
         }
 
+        val tabsNameKey = getPreferenceKey(R.string.pref_key_sync_tabs)
+        findPreference<CheckBoxPreference>(tabsNameKey)?.apply {
+
+            this.isVisible = shouldShowSyncTabs
```

Comment:
I don't think we need to adjust the visibility here, `updateSyncEngineStates` should take care of this.